Assignment/Program.cs

```diff
@@ -1,7 +1,49 @@
-﻿namespace Assignment {
-   internal class Program {
-      static void Main (string[] args) {
-         Console.WriteLine ("Hello world!");
+﻿for (; ; ) {
+   Console.Write ("Enter a number : ");
+   string decimalNumber = Console.ReadLine ();
+   if (int.TryParse (decimalNumber, out int decimalNum)) {
+      //Decimal to Binary
+      int dec = decimalNum, i = 0;
+      var l = new List<int> ();
+      while (dec > 0) {
+         int rem = dec % 2;
+         l.Add (rem);
+         dec /= 2;
       }
-   }
+      Console.Write ("Binary : ");
+      for (i = l.Count - 1; i >= 0; i--)
+         Console.Write (l[i]);
+      Console.WriteLine ();
+
+      //Decimal to Hexadecimal
+      int dec2 = decimalNum, k = 0;
+      var hex = new List<string> ();
+      Dictionary<int, char> dict = new ()
+      {
+   { 10, 'A'},
+   {11, 'B' },
+   {12, 'C' },
+   {13, 'D' },
+   {14, 'E' },
+   {15, 'F' }
+};
+      var a = new List<int> ();
+      while (dec2 > 0) {
+         int rem = dec2 % 16;

```

Comment:
Some code repeats for binary and hex. So you can refactor and simplify both the conversions in a single method which can take parameters to say if you want binary or hex.

Reply:
ok